client/my-sites/domains/domain-management/edit-contact-info/form-card.jsx

```diff
@@ -98,6 +116,22 @@ class EditContactInfoFormCard extends React.Component {
 		} );
 	}
 
+	sanitize = ( fieldValues, onComplete ) => {
```

Comment:
Mmm, this makes it hard to type a space actually. If I type slowly, and do `Andrija ` and wait a bit, it gets trimmed. I know you went above and beyond with this one, but would stick with e-mail `trim` as e-mails don't contain space. :)

Not sure about the `deburr`. Is this a problem actually?